src/main/kotlin/com/demonwav/mcdev/i18n/I18nEditorNotificationProvider.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Minecraft Dev for IntelliJ
+ *
+ * https://minecraftdev.org
+ *
+ * Copyright (c) 2017 minecraft-dev
+ *
+ * MIT License
+ */
+
+package com.demonwav.mcdev.i18n
+
+import com.demonwav.mcdev.i18n.lang.I18nFileType
+import com.demonwav.mcdev.i18n.lang.gen.psi.I18nTypes
+import com.intellij.openapi.command.WriteCommandAction
+import com.intellij.openapi.editor.colors.EditorColors
+import com.intellij.openapi.editor.colors.EditorColorsManager
+import com.intellij.openapi.fileEditor.FileEditor
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.Key
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.PsiManager
+import com.intellij.ui.EditorNotificationPanel
+import com.intellij.ui.EditorNotifications
+import com.intellij.util.ui.UIUtil
+import java.awt.Color
+
+class I18nEditorNotificationProvider(private val project: Project) : EditorNotifications.Provider<I18nEditorNotificationProvider.InfoPanel>() {
+    private var show: Boolean = true
+
+    override fun getKey() = KEY
+
+    override fun createNotificationPanel(file: VirtualFile, fileEditor: FileEditor): InfoPanel? {
+        if (!show || file.fileType != I18nFileType || file.nameWithoutExtension.toLowerCase() == I18nConstants.DEFAULT_LOCALE) {
+            return null
+        }
+
+        val defaultProperties = project.findDefaultProperties(scope = Scope.PROJECT, domain = I18nElementFactory.getResourceDomain(file))
+        val properties = project.findProperties(file = file, scope = Scope.PROJECT)
+        val defaultKeys = defaultProperties.map { it?.key }.toMutableSet()
+        val keys = properties.map { it?.key }.toMutableSet()
```

Comment:
Do you modify this anywhere? As far as I can tell, only `defaultKeys` is modified. No need to make this mutable.